packages/calling/calling/src/CallingClient/CallingClient.ts

```diff
@@ -195,40 +89,73 @@ export class CallingClient extends Eventing<CallingClientEventTypes> implements
     if (!this.sdkConnector.getWebex()) {
       SDKConnector.setWebex(webex);
     }
-
+    this.mutex = new Mutex();
     this.webex = this.sdkConnector.getWebex();
 
     this.sdkConfig = config;
     const serviceData = this.sdkConfig?.serviceData?.indicator
       ? this.sdkConfig.serviceData
       : {indicator: ServiceIndicator.CALLING, domain: ''};
 
+    const logLevel = this.sdkConfig?.logger?.level ? this.sdkConfig.logger.level : LOGGER.ERROR;
     this.validateServiceData(serviceData);
 
-    this.registration = createRegistration(this.webex, serviceData);
+    this.registration = createRegistration(
+      this.webex,
+      serviceData,
+      this.mutex,
+      (event: EVENT_KEYS, deviceInfo?: IDeviceInfo, clientError?: CallingClientError) => {
+        switch (event) {
+          case EVENT_KEYS.REGISTERED:
+            if (deviceInfo) {
+              this.emit(event, deviceInfo);
+            }
+            break;
+          case EVENT_KEYS.UNREGISTERED:
+          case EVENT_KEYS.RECONNECTED:
+          case EVENT_KEYS.RECONNECTING:
+            this.emit(event);
+            break;
+          case EVENT_KEYS.ERROR:
+            if (clientError) {
+              this.emit(event, clientError);
+            }
+            break;
+          default:
+            break;
+        }
+      },
+      logLevel
+    );
     this.callManager = getCallManager(this.webex, serviceData.indicator);
-    this.metricManager = getMetricManager(this.webex, serviceData.indicator);
 
     this.mediaEngine = Media;
+
     this.primaryMobiusUris = [];
     this.backupMobiusUris = [];
-    this.rehomingIntervalMin = DEFAULT_REHOMING_INTERVAL_MIN;
-    this.rehomingIntervalMax = DEFAULT_REHOMING_INTERVAL_MAX;
-    this.failback429RetryAttempts = 0;
-    this.clientError = createClientError('', {}, ERROR_TYPE.DEFAULT, MobiusStatus.DEFAULT);
-    this.isRegistered = MobiusStatus.DEFAULT;
-    this.reconnectPending = false;
+
+    this.registration.setRegistrationStatus(MobiusStatus.DEFAULT);
     this.registerSessionsListener();
-    const logLevel = this.sdkConfig?.logger?.level ? this.sdkConfig.logger.level : LOGGER.ERROR;
 
-    this.mutex = new Mutex();
     log.setLogger(logLevel);
-    /* Better to run the timer once rather than after every registration */
-    this.detectNetworkChange();
+
     this.incomingCallListener();
     this.registerCallsClearedListener();
   }
 
+  public getRegistration(): IRegistrationClient {
```

Comment:
Does this have to be public? When would a developer need to get the registration client object?